packages/plugin-compiler/src/main/kotlin/io/realm/compiler/IrUtils.kt

```diff
@@ -107,8 +109,20 @@ fun IrPluginContext.blockBody(
 val ClassDescriptor.isRealmObjectCompanion
     get() = isCompanionObject && (containingDeclaration as ClassDescriptor).hasRealmModelInterface
 
-val ClassDescriptor.hasRealmModelInterface
-    get() = getSuperInterfaces().firstOrNull { it.fqNameSafe == FqNames.REALM_MODEL_INTERFACE } != null
+val ClassDescriptor.hasRealmModelInterface: Boolean
+    get() {
+        // Using PSI to find super types to avoid cyclic reference (see https://github.com/realm/realm-kotlin/issues/339)
```

Comment:
It's unlikely since the extension is called on a class descriptor that was resolved as a reference `KtClass` from a PSI/AST node first place ... anyway even if the PSI is null the `hasRealmObjectAsSuperType` property default to `false` so it won't be processed as a `RealmObject`